Civi/Token/TokenRow.php

```diff
@@ -188,11 +188,17 @@ public function customToken($entity, $customFieldID, $entityID) {
       'id' => $entityID,
     ]);
     $fieldValue = \CRM_Utils_Array::value($customFieldName, $record, '');
-
+    $originalValue = $fieldValue;
     // format the raw custom field value into proper display value
     if (isset($fieldValue)) {
       $fieldValue = (string) \CRM_Core_BAO_CustomField::displayValue($fieldValue, $customFieldID);
     }
+    // This is a bit of a clumsy wy of detecting a link field but if you look into the displayValue
+    // function you will understand.... By assigning the url as a plain token the text version can
+    // use it as plain text (not html re-converted which kinda works but not in subject lines)
+    if (is_string($fieldValue) && is_string($originalValue) && strpos($fieldValue, '<a href') !== FALSE && strpos($originalValue, '<a href') === FALSE) {
```

Comment:
I'm not sure I do understand. Wouldn't it be cleaner just to check the custom field html_type directly from the custom field ID?

Reply:
@larssandergreen yeah - that would be an extra lookup & we don't have a really easy way to do it at that point cos we need to convert between apiv4 & v3 style keys to get from the class & the function called is awful - hence I went with punting it for now with a long comment

Reply:
Sure, fair enough.
Do textareas work OK getting nl2br-ed in formatDisplayValue and then converted back for text, however that works?

Reply:
Seems OK on testing.

Reply:
FYI @eileenmcnaughton we no longer have to rely on `strpos` now that we have shims in place for PHP8, we can safely use `str_contains` and `str_starts_with` etc.
```suggestion
    if (is_string($fieldValue) && is_string($originalValue) && str_contains($fieldValue, '<a href') && !str_contains($originalValue, '<a href')) {
```

Reply:
@colemanw that's cool - might take me a while to adapt to that  since my non-civi code might not have the shim yet (not sure I want to push a change for this PR though)

Reply:
Just an FYI